integration_tests/preferences.integration.test.tsx

```diff
@@ -74,3 +76,32 @@ describe("User Journey: Update Preferences", () => {
     });
   });
 });
+
+describe("User Journey: Can Navigate back if backTo is set", () => {
+  beforeEach(async () => {
+    // This any annoys me, but otherise I have to mock the entire module.
+    vi.mocked(useSearchParams as any).mockImplementation(() => ({
+      get: (key: string) =>
+        key === "backTo" ? "/exercise/barbell_deadlift" : null,
+      toString: () => "backTo=/exercise/barbell_deadlift",
+    }));
+  });
+
+  afterEach(() => {
+    vi.resetAllMocks();
+  });
+
+  it("should show a Cancel button that navigates to the backTo location if present", async () => {
+    let page = await PreferencesPage();
+    await act(async () => render(page));
+
+    await act(async () => {
+      // There should be a Cancel button
+      const cancelButton = await waitFor(() =>
+        screen.getByTestId(TestIds.Preferences_CancelButton),
+      );
+      const path = new URL((cancelButton as HTMLAnchorElement).href).pathname;
+      expect(path).toEqual("/exercise/barbell_deadlift");
```

Comment:
It's nice to be able to do a simple test like this because we don't have to verify that like, anchors work in html.